include/ccf/crypto/verifier.h

```diff
@@ -14,7 +14,7 @@ namespace ccf::crypto
   class Verifier
   {
   protected:
-    std::shared_ptr<PublicKey> public_key;
+    KeyVariant<RSAPublicKeyPtr, ECPublicKeyPtr> public_key{nullptr};
 
   public:
     Verifier() : public_key(nullptr) {}
```

Comment:
```suggestion
    std::variant<RSAPublicKeyPtr, ECPublicKeyPtr> public_key;

  public:
    Verifier() = default;
```

I don't think we need to define a `KeyVariant` wrapper type, since its only used in 2 places, neither public, and has unclear semantics (with the additional `InvalidKeyPtr` type). This and `jwt_auth.cpp` can directly use a `std::variant<RSAPublicKeyPtr, ECPublicKeyPtr>`, and no other code should need to change.

Reply:
We could have an ECRSAKeyVariant, but I agree that an unspecified KeyVariant that we might be tempted to extend doesn't seem useful. If it gets picked up, extension will break all usage unnecessarily.